### ISSUE TITLE
EVENT: OnEventHit is bugged

### ISSUE THREAD

**Issue by nicolacattaneo** [FlightControl-Master/MOOSE#516]
in mission "EVT-101 - UNIT OnEventHit Example", while the Tank:OnEventHit funtion works properly when the tank is hitten, Plane:OnEventHit fires also when the plane hits the tank, resulting in two messages being desplayd.
In your youtube video, the same code works perfectly.
I'm working on a dogfight mission, using this script, and it's currently impossible to understand who is hitting who!

**Comment by FlightControl-Master** [comment]
That should not happen. Will check this when back.

**Comment by FlightControl-Master** [comment]
@Delta-99 looks like the tests weren't that okay ...
I will do an investigation. If I find something that is not good, i will report.
And then we'll need to go for a fix.
The reason of this communication is that the fix may break some of your code.

**Comment by Delta-99** [comment]
Thanks for letting me know. Will follow any fixes.

**Comment by FlightControl-Master** [comment]
@nicolacattaneo 

Hi Nicola,

I have retested the "hit" test mission with my latest version. A plane fires a missile and hits a tank. It generates me the following log in dcs.log...

```
00522.599 INFO    SCRIPTING:    882(    13)/E:               EVENT00000.onEvent({[1]="S_EVENT_HIT",[2]={["IniGroupName"]="Plane",["IniUnit"]={["UnitName"]="Plane",["ClassName"]="UNIT",},["TgtCoalition"]=2,["WeaponName"]="weapons.missiles.Vikhr_M",["TgtCategory"]=2,["Weapon"]={["id_"]=16778240,},["TgtDCSGroupName"]="Tank",["TgtDCSUnit"]={["id_"]=16777728,},["TgtObjectCategory"]=1,["TgtUnit"]={["UnitName"]="Tank",["ClassName"]="UNIT",},["IniGroup"]={["GroupName"]="Plane",["ClassName"]="GROUP",},["target"]=,["IniUnitName"]="Plane",["time"]=43235.88,["IniDCSGroupName"]="Plane",["IniDCSGroup"]={["id_"]=79,},["IniDCSUnitName"]="Plane",["TgtGroupName"]="Tank",["IniCoalition"]=1,["id"]=2,["TgtDCSGroup"]={["id_"]=80,},["IniObjectCategory"]=1,["IniTypeName"]="Su-25T",["IniCategory"]=0,["initiator"]={["id_"]=16777472,},["TgtDCSUnitName"]="Tank",["TgtUnitName"]="Tank",["TgtTypeName"]="M-2 Bradley",["TgtGroup"]={["GroupName"]="Tank",["ClassName"]="GROUP",},["IniDCSUnit"]=,["weapon"]=,},[3]="Plane",[4]="Tank",[5]=1,})
00522.599 INFO    SCRIPTING:    932(    13)/E:               EVENT00000.onEvent({[1]="Calling OnEventHit for Class ",[2]="UNIT#000000020",[3]=3,})
```

Only one Hit event log line has been written into the dcs.log file.
It correctly shows in the log the **initiator** and collected **Ini**-fields populated with the "Plane" group properties, and the **target** and collected **Tgt**-fields populated with the "Tank" group properties.

So, reading again the initial text of the bug report, this sentence brings my attention:
"_Plane:OnEventHit fires also when the plane hits the tank, resulting in two messages being desplayd._"
I am a bit confused by this sentence. I only see one message being displayed.

So I have a few questions:
1. What moose.lua version are you using?
2. What dcs version are you running this on (1.5.6 or 2.1.x)?

Maybe post your script and mission on our slack community channels. So we can have a more detailed look. Are you on slack and what is your user-id on slack?

Sven

**Comment by nicolacattaneo** [comment]
I'm running dcs 1.5.6 and moose 2.0 (download 4 days ago) and I'm running  the example mission, without any modification to the code. 
As soon as the tank is hit, both the tank and the plane send the "I got hit" messege. This happens in-game, actually I didn't check the log. 
Probably I messed up somewhere, but I  have no idea how. I'll repeat everything from the start this evening. 
In all the tests the su-25 fires and break when it's 2.5/3 nm away from the tank, so it's unlikely  that it actually gets hit by the Bradley. 
Anyway, thank you for the support.
